### DISCUSSION TITLE
How to prevent obsolete msp patches from installing

### DISCUSSION THREAD

**Discussion by joelbudreau** [wixtoolset/issues#6280] [Questions] [answered]
I’ve got an MSI that I’m going to generate .msp patches for as hotfixes are necessary. The patches will be cumulative (like the following)…

Release 1.0.0 MSI
Release Hotpatch 1 (includes bug fix A)
Release Hotpatch 2 (includes bug fix A and B)

If a customer installs the MSI + hp1 + hp2 and then accidentally tries to re-install hp1 it seems like msiexec will consider hp1 and hp2, notice that hp1 is obsolete, and simply re-install the MSI + hp2 patch. I’d like it to actually exit with some error saying that hp1 is obsolete and not perform any installation. How would this be possible?

**Comment by robmen** [comment, marked as answer]
Not that this is an answer, but I would recommend against trying to fight the behavior here. The Windows Installer's design supports scenarios like Patch1 is a Day0 fix and Patch2 is a bug fix plus the Day0 fix. Allowing the user to install Patch1 while Patch2 is present ensures the user can remove Patch2 (should the bug fix break them) but still be protected for the Day0 fix.

In the end, I expect fighting the behavior is going to be more trouble than it's worth.

**Reply by joelbudreau** [reply]
Thanks for the reply
